### PR TITLE
Add makefile rules to generate videos and merge lecture slides

### PR DESCRIPTION
**Pull request by teodutu**
- Merging slides from multiple files is done using [`Markdown-PP`](https://github.com/amyreese/markdown-pp)
- The videos are generated using `ffmpeg` when the including makefile defines the variable `TARGETS`.
Each member of this list must be a subfolder inside the `media/` directory that contains `.svg` images.
These images are merged with `ffmpeg` to produce a `.gif` video.

### CODE REVIEW

File: content/common/makefile/slides.mk [hunk 1]
```diff
@@ -1,14 +1,22 @@
 RVMD = reveal-md
```
Comment (razvand) [inline content/common/makefile/slides.mk:2]: I would use:
```
SLIDES ?= slides.mdpp
MEDIA_DIR ?= media
SITE ?= _site    # Is this configurable?
```

And I would use these variables throughout the `slides.mk` file. With the developer being able to define them in the including Makefile.

**Review by razvand — CHANGES_REQUESTED** [review]
Thanks, @teodutu . Please see my comments.

Apart from those, please add a simple test: 2-3 `.md` files to be joined together in `slides.md` via `markdown-pp` and also 2-3 files to be converted into a `...-generated.gif` file via `ffmpeg`.

File: content/common/makefile/slides.mk (hunk 1 above)
Comment (teodutu) [reply content/common/makefile/slides.mk:10]: The `_site` directory is configurable. I used it as a variable.

**Comment by teodutu** [comment]
I've also added an `open` rule that opens the file `$(SITE)/index.html` in the default browser using `xdg-open`.

**Comment by razvand** [comment]
@teodutu , looks good. Please add a simple test: 2-3 `.md` files to be joined together in `slides.md` via `markdown-pp` and also 2-3 files to be converted into a `...-generated.gif` file via `ffmpeg`.

File: content/common/makefile/slides.mk [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
-	$(RVMD) $< --static $@
+html: $(SITE)
+
+$(SITE): $(SLIDES)
+	$(MDPP) $< -o $(SLIDES_OUT)
+	$(RVMD) $(SLIDES_OUT) --static $@
+
+videos:
+	for TARGET in $(TARGETS); do \
+		$(FFMPEG) -framerate 0.5 -f image2 -y \
```
Comment (razvand) [inline content/common/makefile/slides.mk:23]: I find it easier to use `convert` as done [here](https://github.com/systems-cs-pub-ro/uso-lab-book/blob/task-admin-remote/chapters/task-admin/img/reverse-ssh/Makefile#L14). It doesn't have the strict syntax of input files that `ffmpeg` does.

File: content/common/makefile/slides.mk [hunk 3]
```diff
@@ -1,14 +1,33 @@
...
+			-i "$(MEDIA_DIR)/$$TARGET/$$TARGET-%d.svg" $(MEDIA_DIR)/$$TARGET-generated.gif; \
+	done
+
+open: $(SITE)
+	$(OPEN) $</index.html
```
Comment (razvand) [inline content/common/makefile/slides.mk:28]:
```suggestion
	$(OPEN) $</index.html
```

**Review by razvand — CHANGES_REQUESTED** [review]
Thanks, @teodutu , this looks good and it works.

Apart from my comments below, please see the following:
* Add a `.gitignore` file in `common/test/slides/media/block/`, ignoring the generated GIF file. This should be common to all media folders that generated GIFs.
* I would rename `c-makefiles/` to `c/`. Similar to `slides/`. You don't use `slide-makefiles/`.

File: content/common/makefile/slides.mk (hunk 2 above)
Comment (teodutu) [reply content/common/makefile/slides.mk:23]: Not sure `convert` is what we want.
- It renders the `.gif` at a low quality despite using, for example, the `-weight Thin` argument for thinner text. Below are two samples. The first is created using `convert` and the second is created using `ffmpeg`. The latter looks just like the `.svg` files from which it was generated.

image

image

- It issues the errors below when trying to compile the slides for the compute lecture, despite generating the `.gif`s.

Comment (razvand) [reply content/common/makefile/slides.mk:23]: Gotcha, we'll stick with ffmpeg.

**Review by razvand — APPROVED** [review]
Thanks, @teodutu , this is good.

Please update the commit message to say `makefile: Create test ...` instead of `makefile: Create tests ...` (i.e. use singular).

**Comment by teodutu** [comment]
@razvand Done

**Pull request merged**